packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -463,10 +472,11 @@ export async function downloadSnapshot(
         signal: controller?.signal,
         method: "POST",
     };
-    // For now we are keeping both json and ms-fluid values in accept header while fetching as we want to let
-    // the server decide what format it wants to send to the client as we roll it out slowly.
-    if (fetchBinarySnapshotFormat) {
+    // Decide what snapshot format to fetch as per the feature gate.
+    if (snapshotFormatFetchType?.valueOf() === SnapshotFormatSupportType.JsonAndBinary) {
```

Comment:
why do you need to do valueOf()? I think if you drop it, it should just work, no?

Reply:
It would be great to have a switch statement instead of series of IFs.

Reply:
Removed and used switch statement.